tensorflow_model_optimization/python/core/sparsity_tf2/schedule_test.py

```diff
@@ -14,30 +14,361 @@
 # ==============================================================================
 """Tests for the key functions in pruner library."""
 
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-
-# import g3
-
 from absl.testing import parameterized
-import numpy as np
 import tensorflow as tf
+import functools
 
-# TODO(b/139939526): move to public API.
 from tensorflow.python.keras import keras_parameterized
-from tensorflow_model_optimization.python.core.keras import compat
-from tensorflow_model_optimization.python.core.sparsity_tf2 import pruner
-from tensorflow_model_optimization.python.core.sparsity.keras import pruning_schedule
-from tensorflow_model_optimization.python.core.sparsity.keras import pruning_utils
+from tensorflow_model_optimization.python.core.sparsity_tf2 import schedule
 
 test = tf.test
 
-
 class ScheduleTest(test.TestCase, parameterized.TestCase):
+  """Test to verify custome Schedule behaviour for step parameters.
+  """
+
+  def setUp(self):
+    super(ScheduleTest, self).setUp()
+    self.decay_steps = tf.Variable(100) # number of steps to decay for
+    self.lr_decay_fn = tf.keras.experimental.CosineDecay(0.01, self.decay_steps)
+    self.lr_optimizer = functools.partial(tf.keras.optimizers.SGD, learning_rate=self.lr_decay_fn)
+    self.optimizer = tf.keras.optimizers.SGD()
+
+  def _construct_schedule(
+      self, schedule_type, begin_step, end_step, frequency=10, k=3, 
+      optimizer=None, use_default_optimizer=True, clipnorm=None, clipvalue=None):
+    # Uses default values for sparsity. We're only testing begin_step, end_step
+    # and frequency here as a basic requirement.
+    # Other variants of the Constant schedule may have extra parameters tested.
+    # use_default_optimizer is False if we are testing optimizer function
+    initial_drop_fraction = tf.Variable(0.5) # must be a Variable unlike the orthodox sparsity float
```

Comment:
Why must initial_drop_fraction be a variable? can we rewrite the tests to not use a variable here?

Or if the schedules require it to be a schedule, that's a little problematic (again due to serialization/deserialization challenges) & needs to be explicitly documented

Reply:
We talked about this offline: the tests should avoid using `tf.variable`s except in the specific case of a test checking dynamicism when the drop fraction variable is changed